File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py

def solution(numbers, target):
    answer = 0
    def backtracking(sm, n):
        nonlocal answer
        if n == len(numbers):
            if sm == target:
                answer += 1
            return
        backtracking(sm + numbers[n],n+1)
        backtracking(sm - numbers[n],n+1)

    backtracking(0,0)

    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;
int answer = 0;

void dfs(vector<int> numbers, int target, int sm, int n) {
    if (n == numbers.size()){
        if (sm == target){
            answer++;
        }
        return;
    }
    dfs(numbers,target,sm+numbers[n],n+1);
    dfs(numbers,target,sm-numbers[n],n+1);
}
    
    
int solution(vector<int> numbers, int target) {
    dfs(numbers,target,0,0);
    return answer;
}